root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -15,6 +15,106 @@ import {
   Checker,
 } from '../../edit/URLCleanup';
 
+/*
+ * This file tests the cleanups and autoselect / restrictions defined in
+ * root/static/scripts/tests/Control/URLCleanup.js
+ *
+ * The main part of the file is the testData object, which contains
+ * the expected results for all the tests. The following properties
+ * are supported in testData:
+ *
+ * input_url:
+ *      The raw URL "entered by the user" (a string).
+ *
+ *      This is always mandatory.
+ *
+ * input_entity_type:
+ *      The entity type selected by the user (a string, such as 'release').
+ *      This is important because the same URL can be assigned to different
+ *      relationship types (or blocked altogether) depending on the selected
+ *      entity type.
+ *
+ *      Required by 'expected_relationship_type', 'expected_error'
+ *      and 'limited_link_type_combinations'. Forbidden if none are present.
+ *
+ * expected_relationship_type:
+ *      The relationship type (or types) we expect the URL
+ *      to get autoselected to. Either a string, such as 'downloadfree',
+ *      or an array of strings, such as ['downloadfree', 'streamingfree'].
```

Comment:
I'd perhaps mention what these strings are or where they come from: they're keys from the `LINK_TYPES`/`CLEANUPS` objects in the code.  Otherwise it'd be reasonable to assume these are link_type names.

Reply:
Done in 13d9e10